ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js

```diff
@@ -0,0 +1,128 @@
+        const name = guidePath.substring(guidePath.lastIndexOf('/') + 1)
+        return name + '(' + formPath + ')';
+    }
+
+    const _getEventInfo = (_eventName, _fieldName, _fieldType, _panelName,_formName ) => {
+        // local variables to be used in abandon event which can be triggered anytime
+        formName = _formName;
+        fieldName = _fieldName;
+        fieldType = _fieldType;
+        panelName = _panelName;
+        return {
+            formTitle: _formName,
+            fieldTitle: _fieldName,
+            fieldType: _fieldType,
+            panelTitle: _panelName,
+            eventName: _eventName
+        }
+    }
+
+    const registerGuideBridgeEvents = () => {
+        window.guideBridge.connect(()=>{
+            formName = getFormName();
+
+            dataLayer.push({
+                event: FASTTRACK_ANALYTICS_EVENT,
+                eventInfo: {
+                    formTitle: formName,
+                    eventName: FormEvents.RENDER
+                }
+            });
+
+            guideBridge.on('elementFocusChanged', function(event){
+                dataLayer.push({
+                    event: FASTTRACK_ANALYTICS_EVENT,
+                    eventInfo: _getEventInfo(FormEvents.FIELD, event.detail.fieldLabel, event.detail.fieldType, event.detail.panelTitle, formName)
+                });
+            });
+
+            guideBridge.on('elementHelpShown', function(event){
+                dataLayer.push({
+                    event: FASTTRACK_ANALYTICS_EVENT,
+                    eventInfo: _getEventInfo(FormEvents.HELP, event.detail.fieldLabel, event.detail.fieldType, event.detail.panelTitle, formName)
+                });
+            });
+
+            guideBridge.on('elementErrorShown', function(event){
+                dataLayer.push({
+                    event: FASTTRACK_ANALYTICS_EVENT,
+                    eventInfo: _getEventInfo(FormEvents.ERROR, event.detail.fieldLabel, event.detail.fieldType, event.detail.panelTitle, formName)
+                });
+            });
+
+            guideBridge.on('submitStart', function(event){
+                let eventInfo = _getEventInfo(FormEvents.SUBMIT, event.detail.fieldLabel, event.detail.fieldType, event.detail.panelTitle, formName)
+                eventInfo.timeSpentOnForm = ((new Date()).getTime() - startTime)/1000;
+                dataLayer.push({
+                    event: FASTTRACK_ANALYTICS_EVENT,
+                    eventInfo: eventInfo
+                });
+            });
+        });
+    }
+
+
+    const onDocumentReady = () => {
+        const dataLayerEnabled = document.body.hasAttribute("data-cmp-data-layer-enabled");
+        dataLayer = (dataLayerEnabled) ? window.adobeDataLayer = window.adobeDataLayer || [] : undefined;
+
+        if (dataLayerEnabled) {
+            registerGuideBridgeEvents();
+
+            window.addEventListener('beforeunload', function(){
```

Comment:
question: would this event be still triggered for successful form submissions? 

I think we might want to ensure that once a form has been submitted we don't mark it as abondened.